message.go

```diff
@@ -139,24 +138,18 @@ func (m *Message) WriteTo(w io.Writer) (int64, error) {
 	return total, nil
 }
 
-// DecodeMessage deseralizes data (as []byte) and creates a new Message
+// DecodeMessage deserializes data (as []byte) and creates a new Message
 func DecodeMessage(b []byte) (*Message, error) {
 	var msg Message
 
-	msg.Timestamp = int64(binary.BigEndian.Uint64(b[:8]))
-	msg.Attempts = binary.BigEndian.Uint16(b[8:10])
-
-	buf := bytes.NewBuffer(b[10:])
-
-	_, err := io.ReadFull(buf, msg.ID[:])
-	if err != nil {
-		return nil, err
+	if len(b) < 11+MsgIDLength {
```

Comment:
so you're referring to https://github.com/nsqio/nsq/blob/fa56e2fda80c08f1428b9139f516b9aadac117c3/nsqd/message.go#L69-L92

1) i guess we could argue for days about this, but i'm personally not a big fan of having part of the numbers hardcoded in the function and part defined elsewhere. if `minValidMsgLength` ever changes then i can guarantee you this function will change as well. so in my mind it makes more sense to keep it all together in one place.

2) that function allows empty bodies. whether nsqd allows empty bodies or not, i actually don't care much (though we should obviously follow the same rules everywhere). i did this check cause i thought `b[10+16:]` would have an index out of bounds if len was 26, but turns out i'm wrong about that